### ISSUE TITLE
[Remote Store] _cat/recovery APIs provides inconsistent results

### ISSUE THREAD

**Issue by Bukhtawar** [opensearch-project/OpenSearch#12047]
### Describe the bug

1. When compared with total initialising shards in `_cluster/health` API, the `_cat/recovery?active_only` shows an inconsistent count of recoveries in progress.
2. The translog download step doesn't populate the translog recovery stats

### Related component

Storage:Remote

### Expected behavior

Consistent API results

### Additional Details

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/peternied) [2](https://github.com/lukas-vlcek/) [3](https://github.com/andrross) [4](https://github.com/CEHENKLE) [5](https://github.com/dbwiddis) [6](https://github.com/reta/) [7](https://github.com/anasalkouz) [8](https://github.com/aggarwalShivani)]
@Bukhtawar Thanks for filing this issue, this is a very confusing experience and it would be good to address.

**Comment by lukas-vlcek** [comment]
Hi, I would like to take this ticket.

While conducting my investigation, I welcome your insights and recommendations on specific areas to focus on.

**Comment by Bukhtawar** [comment]
Yes I believe this should be reproducible on a multi-node setup, hosting shards of few 100MBs, where we exclude IP of one node and trigger a relocation process of shards on the excluded node.
Then compare the output of `_cluster/health`  "intialising/relocating" count and `_cat/recovery?active_only` to see the discrepancy in count.

**Comment by lukas-vlcek** [comment]
@Bukhtawar Thanks!
Do you think you can elaborate bit on "exclude IP of one node". Do you mean exclude the node from a shard allocation?

Would the following scenario be a good candidate?

Imagine a two node cluster, `Node 1` having three primary shards, `Node 2` being empty.

```mermaid
flowchart TB
    Primary_A
    Primary_B
    Primary_C
    subgraph "Node 1"
    Primary_A
    Primary_B
    Primary_C
    end
    subgraph "Node 2"
    end
```
Next, we exclude the `Node 1` from a shard allocation:

```curl
PUT _cluster/settings
{
  "persistent" : {
    "cluster.routing.allocation.exclude._ip" : "_Node 1 IP_"
  }
}
```
This should trigger replication of all shards from `Node 1` to `Node 2`.

```mermaid
flowchart TB
    Primary_A--"Replicating"-->Replica_A
    Primary_B--"Replicating"-->Replica_B
    Primary_C--"Replicating"-->Replica_C
    subgraph "Node 1"
    Primary_A
    Primary_B
    Primary_C
    end
    subgraph "Node 2"
    Replica_A
    Replica_B
    Replica_C
    end
```
Now, while shards are being replicated, we can request `_cluster/health` and `_cat/recovery?active_only` (as discussed previously) and that should give us inconsistent counts, correct?

**Comment by lukas-vlcek** [comment]
Hi @Bukhtawar

I was looking at this and I found that the following integration test is already testing something very similar:
```
./server/src/internalClusterTest/java/org/opensearch/indices/recovery/IndexRecoveryIT.java
```
For example it has a test called `testRerouteRecovery()` that uses the following scenario:
1. It starts a cluster with a single node (A)
2. It creates a new index with single shard and no replicas
3. Then it adds new node (B) to the cluster
4. Then is slows down recoveries (using `RecoverySettings.INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING`)
5. Then it forces relocation of the shard using "admin.cluster.reroute" request (so a bit different strategy than discussed above)
6. It check for count of active (ie. `stage != DONE`) shard recoveries etc...
7. ...

I was experimenting and modified some tests and added "admin.cluster.health" request into them to get initializing and relocating shard counts and so far I was not able to spot/replicate the count discrepancy.

Do you think it can be because the size of the index in the test is quite small (just couple of 100kbs)? Though, the test explicitly makes sure the counts are obtained while the recovery process is throttled and the shard recovery stage is not DONE (in other words the counts are compared while the recovery is still running).

However, there is still another question I wanted to ask. Did you have anything specific in mind when you said:
> The translog download step doesn't populate the translog recovery stats

Can you elaborate on this please?

I will push modification of the test tomorrow so that you can see what I mean.

**Comment by Bukhtawar** [comment]
Adding @sachinpkale for his thoughts as well. Will take a look shortly

**Comment by shourya035** [comment]
@sachinpkale @Bukhtawar This PR is waiting on your inputs. Can you bring this to closure?
